acme/examples/http01_example.py

```diff
@@ -0,0 +1,226 @@
+"""Example ACME-V2 API for HTTP-01 challenge.
+
+Brief:
+
+This a complete usage example of the python-acme API.
+
+Limitations of this example:
+    - Works for only one Domain name.
+    - Performs only HTTP-01 challenge.
+    - Uses ACME-v2
+
+Workflow:
+    - Create account key.
+    - Register account and accept TOS.
+    - Select HTTP-01 within offered challenges by the CA server.
+    - Set up standalone web server.
+    - Create domain private key and CSR.
+    - Issue certificate.
+    - Renew Certificate
+    - Change contact information
+    - Revoke Certificate
+    - Deactivate Account
+"""
+from contextlib import contextmanager
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives.asymmetric import rsa
+import OpenSSL
+
+from acme import challenges
+from acme import client
+from acme import crypto_util
+from acme import errors
+from acme import messages
+from acme import standalone
+import josepy as jose
+
+# Constants:
+
+# This is the staging point for ACME-V2 within Let's Encrypt.
+DIRECTORY_URL = 'https://acme-staging-v02.api.letsencrypt.org/directory'
+
+USER_AGENT = 'python-acme-example'
+
+# Account key size
+ACC_KEY_BITS = 2048
+
+# Certificate private key size
+CERT_PKEY_BITS = 2048
+
+# Domain name for the certificate.
+DOMAIN = 'client.example.com'
+
+# If you are running Boulder locally, it is possible to configure any port
+# number to execute the challenge, but real CA servers will obviously always
```

Comment:
Turns out this isn't obvious, so let's replace "obviously" with "will always use port 80, as described in the ACME specification"

Reply:
Ok, sounds good.